match/lib/match/runner.rb

```diff
@@ -254,6 +254,25 @@ def fetch_provisioning_profile(params: nil, certificate_id: nil, app_identifier:
         end
       end
 
+      if params[:include_all_certificates]
+        # Clearing specified certificate id which will prevent a profile being created with only one certificate
+        certificate_id = nil
+
+        if params[:force_for_new_certificates] && !params[:readonly]
+          if prov_type == :development && !params[:force]
+            force = certificate_count_different?(profile: profile, keychain_path: keychain_path, platform: params[:platform].to_sym)
+          else
```

Comment:
One more question! If `prov_type` is `:development` and `force` is set to `true` already, won't the `"Warning: `force_for_new_certificates` is set but is ignored for non-'development' provisioning profiles."` get printed out?

I think that check for `force` needs to happen somewhere else, correct?

Reply:
Good catch. I think I'll need to wrap it in the `unless force`

Reply:
Fixed. Better to review [w/o space changes](https://github.com/fastlane/fastlane/pull/19340/files?diff=unified&w=1)

Reply:
Moved all checks into separate methods to reduce complexity.